services/consensusalgo/benchmarkconsensus/test/create_blocks_test.go

```diff
@@ -0,0 +1,27 @@
+package test
+
+import (
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/test"
+	"testing"
+)
+
+func TestLeaderCreatesBlocks(t *testing.T) {
+	c := newContext(true)
+	c.consensusContext.When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).AtLeast(1)
+	c.createService()
+	err := test.EventuallyVerify(c.consensusContext)
+	if err != nil {
+		t.Fatal("Did not create block with ConsensusContext:", err)
+	}
+}
+
+func TestNonLeaderDoesNotCreateBlocks(t *testing.T) {
+	c := newContext(false)
+	c.consensusContext.When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).Times(0)
+	c.createService()
+	err := test.ConsistentlyVerify(c.consensusContext)
```

Comment:
`ConsistentlyVerify`? is this your implementation of 'eventually'?

Reply:
There are 2 functions - `eventually` and `consistently`. Ginkgo has both in its API and it turned out I needed both. I also added specialized cases for mocks to reduce boilerplate. The full code is here: https://github.com/orbs-network/orbs-network-go/pull/23/files#diff-9668350988ac1f28905c796e007c35e5